pom.xml

```diff
@@ -49,7 +49,7 @@
         <dep.slice.version>0.38</dep.slice.version>
         <dep.testing-mysql-server-5.version>0.6</dep.testing-mysql-server-5.version>
         <dep.aws-sdk.version>1.11.697</dep.aws-sdk.version>
-        <dep.okhttp.version>3.9.0</dep.okhttp.version>
+        <dep.okhttp.version>4.5.0</dep.okhttp.version>
```

Comment:
Is this absolutely needed?

@tanjialiang, this would break our presto-facebook build. Could you help to check if 4.5 is a version we can use?

Reply:
We can test out if @achen2289 can help us understand the new version is a hard-requirement

Reply:
Yes, I've spent time with @cliandy and this is the only solution we've come upon. Problem is any okhttp version above 4.5.0 has conflicting kotlin std lib dependencies that cause maven enforcer plugin to fail the build.

Reply:
[Require upper bound dependencies error for org.jetbrains.kotlin:kotlin-stdlib-common:1.5.31 paths to dependency are:
+-com.facebook.presto:presto-client:0.278-SNAPSHOT
  +-com.squareup.okhttp3:okhttp:4.10.0
    +-com.squareup.okio:okio-jvm:3.0.0
      +-org.jetbrains.kotlin:kotlin-stdlib-common:1.5.31
and
+-com.facebook.presto:presto-client:0.278-SNAPSHOT
  +-com.squareup.okhttp3:okhttp:4.10.0
    +-org.jetbrains.kotlin:kotlin-stdlib:1.6.20
      +-org.jetbrains.kotlin:kotlin-stdlib-common:1.6.20
]](url)

Reply:
```
Require upper bound dependencies error for org.jetbrains.kotlin:kotlin-stdlib-common:1.5.31 paths to dependency are:
+-com.facebook.presto:presto-client:0.278-SNAPSHOT
  +-com.squareup.okhttp3:okhttp:4.10.0
    +-com.squareup.okio:okio-jvm:3.0.0
      +-org.jetbrains.kotlin:kotlin-stdlib-common:1.5.31
and
+-com.facebook.presto:presto-client:0.278-SNAPSHOT
  +-com.squareup.okhttp3:okhttp:4.10.0
    +-org.jetbrains.kotlin:kotlin-stdlib:1.6.20
      +-org.jetbrains.kotlin:kotlin-stdlib-common:1.6.20
]
```

Reply:
Thanks for confirming @achen2289

Reply:
Are we sure this isn't a local setup something? I downloaded your patch and compiled with 3.9.0. It compiles well....

Reply:
The build works even with 3.9.0. However, okhttp 3.9.0 has issues at runtime due to missing methods. This is why I bumped the version. The build issues I was saying earlier apply to versions of okhttp > 4.5.0.

Reply:
If that is the case, can we put 3.9.0 for this PR? We probably wanna separate them to avoid major regressions. We can have a single PR later on to just bump up the version. Meanwhile, I will review the design of the PR.

BTW, what is the runtime stacktrace of the missing method?

Reply:
Okay. I will push a separate commit to use 3.9.0.

Reply:
```
java.lang.NoSuchMethodError: okhttp3.OkHttpClient$Builder.callTimeout(Ljava/time/Duration;)Lokhttp3/OkHttpClient$Builder;
	at io.opentelemetry.exporter.internal.grpc.GrpcExporterBuilder.build(GrpcExporterBuilder.java:139)
	at io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporterBuilder.build(OtlpGrpcSpanExporterBuilder.java:153)
	at com.facebook.presto.opentelemetry.OpenTelemetryBuilder.build(OpenTelemetryBuilder.java:62)
	at com.facebook.presto.opentelemetry.OpenTelemetryTracer.<clinit>(OpenTelemetryTracer.java:36)
	at com.facebook.presto.opentelemetry.OpenTelemetryTracerProvider.getNewTracer(OpenTelemetryTracerProvider.java:42)
	at com.facebook.presto.server.HttpRequestSessionContext.<init>(HttpRequestSessionContext.java:225)
	at com.facebook.presto.server.protocol.QueuedStatementResource.postStatement(QueuedStatementResource.java:219)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory.lambda$static$0(ResourceMethodInvocationHandlerFactory.java:76)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:148)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:191)
	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$ResponseOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:200)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:103)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:493)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:415)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:104)
	at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:277)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:272)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:268)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:316)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:298)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:268)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:289)
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:256)
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:703)
	at org.glassfish.jersey.servlet.WebComponent.serviceImpl(WebComponent.java:416)
	at org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:370)
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:389)
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:342)
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:229)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:867)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1623)
	at com.facebook.airlift.http.server.AuthenticationFilter.doFilter(AuthenticationFilter.java:69)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1610)
	at com.facebook.airlift.http.server.TraceTokenFilter.doFilter(TraceTokenFilter.java:63)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1610)
	at com.facebook.airlift.http.server.TimingFilter.doFilter(TimingFilter.java:51)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1610)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:540)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:146)
	at org.eclipse.jetty.server.handler.gzip.GzipHandler.handle(GzipHandler.java:703)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:257)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1345)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:480)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1247)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144)
	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:126)
	at org.eclipse.jetty.server.handler.StatisticsHandler.handle(StatisticsHandler.java:174)
	at org.eclipse.jetty.server.handler.HandlerList.handle(HandlerList.java:61)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.Server.handle(Server.java:502)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:364)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:260)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:305)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:118)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:126)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:366)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:765)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:683)
	at java.lang.Thread.run(Thread.java:748)
```